tools/nvqpp-v1/nvqpp_targets.cpp

```diff
@@ -0,0 +1,157 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+#include "nvqpp_targets.h"
+#include "nvqpp_options.h"
+#include <fstream>
+#include <iostream>
+#include <regex>
+
+namespace cudaq {
+TargetPlatformConfig
+Target::generateConfig(const llvm::opt::InputArgList &clOptions,
+                       const std::filesystem::path &targetConfigSearchPath) {
+  TargetPlatformConfig config;
+  // Load from file:
+  const std::string fileName = name + std::string(".config");
```

Comment:
I would like us to figure out how to drop the bash config files and pick up a json or yaml file instead.

Reply:
The difficult thing is how to handle things like `--quantinuum-url`. Maybe the config json file can contribute extra command line arguments that can be picked up and exposed to the driver?

Reply:
What about config files? https://releases.llvm.org/6.0.0/tools/clang/docs/UsersManual.html#configuration-files

Reply:
+1 for LLVM config file :-) 

The difficulty of handling things like `--quantinuum-url` is that we need to make them extensible (new backends can contribute new arguments easily) while also encoding the logic to convert them into a proper `#define` string. 

Looking at Clang/LLVM, it has a combined `options.td` which virtually contains everything. Not sure if this is appropriate for our case, e.g., require new backend implementations to add entries to the argument list.

Reply:
We should make a list of what we suspect backends may need for support (based on our experience so far).